plugins/config_manager.py

```diff
@@ -0,0 +1,250 @@
+                dnfpluginscore.lib.write_raw_configfile(dnf.const.CONF_FILENAME,
+                                                        'main', sbc.substitutions,
+                                                        sbc.cfg.options,
+                                                        sbc.iteritems,
+                                                        sbc.optionobj,
+                                                        modify)
+
+        if self.opts.set_enabled or self.opts.set_disabled:
+            self.opts.save = True
+            modify.append('enabled')
+
+        if self.opts.repo:
+            matched = []
+            for name in self.opts.repo:
+                matched.extend(self.base.repos.get_matching(name))
+        else:
+            matched = self.base.repos.iter_enabled()
+
+        for repo in sorted(matched):
+            print(self.base.output.fmtSection('repo: ' + repo.id))
+            if self.opts.set_enabled and not repo.enabled:
+                repo.enable()
+            elif self.opts.set_disabled and repo.enabled:
+                repo.disable()
+            print(repo.dump())
+            repo_modify = modify[:]
+            if (hasattr(self.cli, 'repo_setopts')
+                    and repo.id in self.cli.repo_setopts):
+                repo_modify.extend(self.cli.repo_setopts[repo.id].items)
+            if self.opts.save and modify:
+                dnfpluginscore.lib.write_raw_configfile(repo.repofile,
+                                                        repo.id,
+                                                        sbc.substitutions,
+                                                        repo.cfg.options,
+                                                        repo.iteritems,
+                                                        repo.optionobj,
+                                                        repo_modify)
+
+    def add_repo(self):
+        """ process --add-repo option """
+
+        # put repo file into first reposdir which exists or create it
+        myrepodir = None
+        for rdir in self.base.conf.reposdir:
+            if os.path.exists(rdir):
+                myrepodir = rdir
+                break
+
+        if not myrepodir:
+            myrepodir = self.base.conf.reposdir[0]
+            dnf.util.ensure_dir(myrepodir)
+
+        for url in self.opts.add_repo:
+            if dnf.pycomp.urlparse.urlparse(url).scheme == '':
+                url = 'file://' + os.path.abspath(url)
+            logger.info(_('Adding repo from: %s'), url)
+            if url.endswith('.repo'):
+                # .repo file - download, put into reposdir and enable it
+                destname = os.path.basename(url)
+                destname = myrepodir + '/' + destname
+                try:
+                    f = dnfpluginscore.lib.urlopen(self, None, url, 'w+')
+                    f.close()
+                    shutil.copy2(f.name, destname)
```

Comment:
Once you call f.close() then the file is deleted so copy2() have nothing to copy. You should pass delete=False to urlopen() to prevent this behaviour.